homepage/tests.py

```diff
@@ -191,3 +191,111 @@ def test_print_details(capsys, review_id, expected_review_details):
+         Course(3, "Course3", False, 1, "N/A", 2, 2.5, 0, 0)]
+    for course in courses_list:
+        course.save()
+
+    return courses_list
+
+
+# creating a new prerequisite log
+@pytest.mark.django_db
+def test_create_new_prerequisite(courses):
+    valid = True
+    preq = Prerequisites(course_id=courses[0], req_course_id=courses[1], req_code=Prerequisites.Req_Code.BEFORE)
+    try:
+        preq.clean_fields()
+        preq.save()
+    except ValidationError:
+        valid = False
+
+    assert valid and Prerequisites.objects.filter(course_id=preq.course_id).count() == 1
+
+
+# creating a new invalid prerequisite log - req_code invalid
+@pytest.mark.django_db
+def test_create_new_invalid_prerequisite(courses):
+    invalid = False
+    preq = Prerequisites(course_id=courses[1], req_course_id=courses[0], req_code=3)
+    try:
+        preq.clean_fields()
+    except ValidationError:
+        invalid = True
+
+    assert invalid
+
+
+@pytest.mark.django_db
+def test_prerequisite_print(courses, capsys):
+    preq = Prerequisites(course_id=courses[1], req_course_id=courses[0], req_code=Prerequisites.Req_Code.BEFORE)
+    preq.save()
+
+    msg = f"req_cour = Course1, desired_cour = Course2, req_code = {Prerequisites.Req_Code.BEFORE}\n"
+    print(preq)
+    assert capsys.readouterr().out == msg
+
+
+@pytest.mark.django_db
+# test for Course2 which has prerequisite - Course1
+def test_get_prerequisites_for_course_with_one_preqs(courses):
+    preq = Prerequisites(course_id=courses[1], req_course_id=courses[0], req_code=Prerequisites.Req_Code.BEFORE)
+    preq.save()
+
+    assert Prerequisites.get_prerequisites_for_course(courses[1]).count() == 1
+
+
+@pytest.mark.django_db
+# test for Course2 which here depends on taking Course1 and Course3 both
+def test_get_prerequisites_for_course_with_mult_preqs(courses):
+    preqs = [Prerequisites(course_id=courses[1], req_course_id=courses[0], req_code=Prerequisites.Req_Code.BEFORE),
+             Prerequisites(course_id=courses[1], req_course_id=courses[2], req_code=Prerequisites.Req_Code.BEFORE)]
+
+    for preq in preqs:
+        preq.save()
+
+    assert Prerequisites.get_prerequisites_for_course(courses[1]).count() == 2
+
+
+@pytest.mark.django_db
+# Course1 has no prerequisites
+def test_get_prerequisites_for_course_with_no_preqs(courses):
+
+    assert not Prerequisites.get_prerequisites_for_course(courses[0]).exists()
+
+
+@pytest.mark.django_db
+# test to see it returns True
+def test_does_course_have_prerequisites_one_preqs(courses):
+    preq = Prerequisites(course_id=courses[1], req_course_id=courses[0], req_code=Prerequisites.Req_Code.BEFORE)
+    preq.save()
+
+    assert Prerequisites.does_course_have_prerequisites(courses[1])
+
+
+@pytest.mark.django_db
+# test to see it returns True
+def test_does_course_have_prerequisites_for_mult_preqs(courses):
+    preqs = [Prerequisites(course_id=courses[1], req_course_id=courses[0], req_code=Prerequisites.Req_Code.BEFORE),

```

Comment:
I can see many tests that have almost the same code. 
did you try to use parameterized?

Reply:
As I don't want it to create the preqs sometimes, I put it inside a fixture